plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -128,6 +200,87 @@ public Page getNextPage()
         }
     }
 
+    @Override
+    public void deleteRows(Block rowIds)
+    {
+        RowBlock rows = resolveRowIdBlock(rowIds);
+        posDeleteSink.appendPage(new Page(rows.getPositionCount(), rows.getChildren().toArray(new Block[0])));
+    }
+
+    @Override
+    public void updateRows(Page page, List<Integer> columnValueAndRowIdChannels)
+    {
+        int batchSize = page.getPositionCount();
+        Block rowIdRawBlock = page.getBlock(columnValueAndRowIdChannels.get(columnValueAndRowIdChannels.size() - 1));
+        RowBlock rowIdBlock = resolveRowIdBlock(rowIdRawBlock);
+        Block[] rowIdFieldBlocks = rowIdBlock.getChildren().toArray(new Block[0]);
+        Block[] posDeleteBlocks = new Block[2];
+        posDeleteBlocks[0] = rowIdFieldBlocks[0];
+        posDeleteBlocks[1] = rowIdFieldBlocks[1];
+        posDeleteSink.appendPage(new Page(batchSize, posDeleteBlocks));
+
+        Block[] allTableColumnBlocks = new Block[allTableColumnChannels.length];
+        for (int i = 0; i < allTableColumnChannels.length; i++) {
+            int allTableColumnChannel = allTableColumnChannels[i];
+            if (allTableColumnChannel > -1) {
+                allTableColumnBlocks[i] = rowIdFieldBlocks[allTableColumnChannel + 2];
+            }
+            else {
+                int updateValueChannelIndex = (-1) - allTableColumnChannel;
+                allTableColumnBlocks[i] = page.getBlock(columnValueAndRowIdChannels.get(updateValueChannelIndex));
+            }
+        }
+
+        updateRowSink.appendPage(new Page(batchSize, allTableColumnBlocks));
+    }
+
+    private RowBlock resolveRowIdBlock(Block rowIds)
+    {
+        RowBlock rows;
+        if (rowIds instanceof RowBlock) {
+            rows = (RowBlock) rowIds;
+        }
+        else if (rowIds instanceof DictionaryBlock) {
+            Block dictionary = ((DictionaryBlock) rowIds).getDictionary();
+            if (dictionary instanceof RowBlock) {
+                rows = (RowBlock) dictionary;
+            }
+            else {
+                throw new TrinoException(GENERIC_INTERNAL_ERROR, "Unexpected rowId dictionary block type " + dictionary);
+            }
+        }
+        else {
+            throw new TrinoException(GENERIC_INTERNAL_ERROR, "Unexpected rowId block type " + rowIds);
+        }
+        return rows;
+    }
+
+    @Override
+    public CompletableFuture<Collection<Slice>> finish()
+    {
+        try {
+            Collection<Slice> slices = new ArrayList<>();
+            if (posDeleteSink != null) {
```

Comment:
if the field is nullable, annotate the field and constructor param with `@Nullable`